### PR TITLE
Update Upgrading Guide for 7.6

### LINKED ISSUE #1563: Update Upgrading Guide for RHSSO 7.6
### Description

Summarize changes for Keycloak 16 to 18

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by andymunro**
Closes #1563

@vmuzikar and @pskopek Please review.

### CODE REVIEW

File: upgrading/topics/rhsso/changes-76.adoc [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  maxUnavailable: 1
+  selector:
+    matchLabels:
+      component: keycloak
+```
+See also the https://kubernetes.io/docs/tasks/run-application/configure-pdb/[Kubernetes Documentation].
+
+=== Keycloak Operator examples including unsupported Metrics extension
+
+Previously, an unsupported metrics extension was added in the example for the creation of the Keycloak CR by the {project_operator}. This has been removed.
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc]: I wouldn't mention this. This mainly concerned the upstream, as in downstream we don't have the affected examples.

File: upgrading/topics/rhsso/changes-76.adoc~ [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+=== Deprecated features in the {project_operator}
+
+With this release, we have deprecated and/or marked as unsupported some features in the {project_operator}. This
+concerns the Backup CRD and the operator managed Postgres Database. For more details, please see the
+link:{installguide_link}#_operator_production_usage[related chapter in the {installguide_name}].
+
+=== Keycloak Operator examples including unsupported Metrics extension
+
+Previously, an unsupported metrics extension was added in the example for the creation of the Keycloak CR by the {project_operator}. This has been removed.
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc~]: I wouldn't mention this. This mainly concerned the upstream, as in downstream we don't have the affected examples.

File: upgrading/topics/rhsso/changes-76.adoc [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+HTTP requests. This change could lead to unexpected use of a proxy server if you have for example the `HTTP_PROXY` variable defined but have no
+explicit proxy mappings specified in your SPI configuration. To prevent {project_name} from using those environment variables,
+you can explicitly create a no proxy route for all requests as `.*;NO_PROXY`.
+
+For more details, see the link:{installguide_link}#_proxy_env_vars[related chapter in the {installguide_name}].
+
+=== Deprecated features in the {project_operator}
+
+With this release, we have deprecated `podDisruptionBudget` field in the Keycloak CR of the https://github.com/keycloak/keycloak-operator[legacy {project_operator}].
+This optional field will be ignored when the Operator is deployed on Kubernetes version 1.25 and higher.
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc]: From the RH-SSO perspective, there's nothing like "legacy operator".

```suggestion
With this release, we have deprecated `podDisruptionBudget` field in the Keycloak CR of {project_operator}.
This optional field will be ignored when the Operator is deployed on OCP 4.12 and newer.
```

File: upgrading/topics/rhsso/changes-76.adoc~ [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+explicit proxy mappings specified in your SPI configuration. To prevent {project_name} from using those environment variables,
+you can explicitly create a no proxy route for all requests as `.*;NO_PROXY`.
+
+For more details, see the link:{installguide_link}#_proxy_env_vars[related chapter in the {installguide_name}].
+
+=== Deprecated features in the {project_operator}
+
+With this release, we have deprecated and/or marked as unsupported some features in the {project_operator}. This
+concerns the Backup CRD and the operator managed Postgres Database. For more details, please see the
+link:{installguide_link}#_operator_production_usage[related chapter in the {installguide_name}].
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc~]: I'm not sure how this got here. This was already affecting 7.5.1.

File: upgrading/topics/rhsso/changes-76.adoc~ [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+existing database before upgrade*. While we did our best to test the consequences of the Liquibase upgrade, some installations could be using specific setup unknown to us.
+
+=== Proxy environment variables
+
+{project_name} now respects the standard `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` environment variables for outgoing
+HTTP requests. This change could lead to unexpected use of a proxy server if you have for example the `HTTP_PROXY` variable defined but have no
+explicit proxy mappings specified in your SPI configuration. To prevent {project_name} from using those environment variables,
+you can explicitly create a no proxy route for all requests as `.*;NO_PROXY`.
+
+For more details, see the link:{installguide_link}#_proxy_env_vars[related chapter in the {installguide_name}].
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc~]: This was [part of 7.5.1](https://issues.redhat.com/browse/KEYCLOAK-19420) but probably was missing from the  release notes. Not sure if we should mention it for 7.6.0.

File: upgrading/topics/rhsso/changes-76.adoc [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+existing database before upgrade*. While we did our best to test the consequences of the Liquibase upgrade, some installations could be using specific setup unknown to us.
+
+=== Proxy environment variables
+
+{project_name} now respects the standard `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` environment variables for outgoing
+HTTP requests. This change could lead to unexpected use of a proxy server if you have for example the `HTTP_PROXY` variable defined but have no
+explicit proxy mappings specified in your SPI configuration. To prevent {project_name} from using those environment variables,
+you can explicitly create a no proxy route for all requests as `.*;NO_PROXY`.
+
+For more details, see the link:{installguide_link}#_proxy_env_vars[related chapter in the {installguide_name}].
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc]: This was [part of 7.5.1](https://issues.redhat.com/browse/KEYCLOAK-19420) but probably was missing from the  release notes. Not sure if we should mention it for 7.6.0.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@andymunro Thanks for the PR!

**Comment by andymunro** [comment]
Thanks, @vmuzikar. Updates made.

File: upgrading/topics/rhsso/changes-76.adoc [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+Closely follow the <<_upgrading,Upgrading Guide>>, specifically of *backing up
+existing database before upgrade*. While we did our best to test the consequences of the Liquibase upgrade, some installations could be using specific setup unknown to us.
+
+=== Deprecated features in the {project_operator}
+
+With this release, we have deprecated `podDisruptionBudget` field in the Keycloak CR of {project_operator}.
+This optional field will be ignored when the Operator is deployed on OCP 4.12 and newer
+
+This optional field will be ignored when the Operator is deployed on Kubernetes version 1.25 and higher.
```
Comment (vmuzikar) [inline upgrading/topics/rhsso/changes-76.adoc]:
```suggestion
This optional field will be ignored when the Operator is deployed on OCP 4.12 and newer.
```

**Comment by andymunro** [comment]
@vmuzikar Can you approve this now?

**Review by vmuzikar — DISMISSED** [review]
@andymunro Thanks for the update! As far as the Operator goes, LGTM. Can't comment on other changes. :)

**Comment by andymunro** [comment]
Broken external links are to 7.6 URLs that are not yet created.

**Review by miquelsi — DISMISSED** [review]

File: topics/templates/document-attributes-product.adoc [hunk 8]
```diff
@@ -1,19 +1,20 @@
 :project_community: false
 :project_product: true
 :project_name: Red Hat Single Sign-On
-:project_versionMvn: 15.0.2.redhat-00001
-:project_versionNpm: 15.0.2.redhat-00001
+:project_versionMvn: 18.0.2.redhat-00001
+:project_versionNpm: 18.0.2.redhat-00001
```
Comment (miquelsi) [inline topics/templates/document-attributes-product.adoc]: 18.0.0

**Comment by andymunro** [comment]
Okay thanks, @miquelsi. Please check and reapprove if okay.

**Review by vmuzikar — APPROVED** [review]

**Review by miquelsi — APPROVED** [review]

**Review by abstractj — APPROVED** [review]

**Pull request merged**